src/target/riscv/batch.c

```diff
@@ -125,33 +133,33 @@ int riscv_batch_run(struct riscv_batch *batch)
 	return ERROR_OK;
 }
 
-void riscv_batch_add_dm_write(struct riscv_batch *batch, unsigned int address, uint64_t data,
+void riscv_batch_add_dm_write(struct riscv_batch *batch, uint64_t address, uint32_t data,
```

Comment:
As stated in the spec [3.1 Debug Module Interface (DMI)]:
> The DMI uses between 7 and 32 address bits.

So maybe it is better to use `uint32_t` for the address?

Reply:
@en-sc Thanks for pointing that out. I missed that the debug spec restricts the DMI address size to 32 bits in the section 3.1 (as you have quoted). 

My reason for uint64_t was the dtmcs.abits field which may contain value up to 63 (but I missed the above restriction).

I doubt that there is any real-world hardware with abits > 32 (what would be the reason for such a large address space for debug?). So we can restrict this, but at the same time we should add a check for abits value, read from the target. I'd prefer to do this in a separate merge request.

Reply:
I have filed an issue for it: https://github.com/riscv-collab/riscv-openocd/issues/1019